scripts/home.js

```diff
@@ -1,34 +1,471 @@
 
-burgerMenu.addEventListener("click",()=>{
-    burgerMenu.classList.toggle("open");
-    if(menuContent.classList.contains("show")){
-        menuContent.classList.remove("show"); 
-        menuContent.classList.add("hide"); 
-    }else{
-        menuContent.classList.remove("hide"); 
-        menuContent.classList.add("show");
-    }
-}); 
-
+burgerMenu.addEventListener("click", () => {
+  burgerMenu.classList.toggle("open");
+  if (menuContent.classList.contains("show")) {
+    menuContent.classList.remove("show");
+    menuContent.classList.add("hide");
+  } else {
+    menuContent.classList.remove("hide");
+    menuContent.classList.add("show");
+  }
+});
 
 // Small screen search icon
-const searchBar = document.querySelector(".search-bar");
+const searchBox = document.querySelector(".search-box");
 const searchIcon = document.querySelector(".search-icon");
-searchInputField =  document.querySelector("#header .search input");
-searchIcon.addEventListener("click",()=>{
-    if(searchBar.classList.contains("hide-search-bar")){
-        searchBar.classList.remove("hide-search-bar");
-        searchInputField.value = "";
-    }else{
-        searchBar.classList.add("hide-search-bar");
-    }
-});  
-
-// prevent page from reload when the search form get submitted 
-document.querySelector("#header .search-bar form").addEventListener("submit",event=>{
-    event.preventDefault(); 
+searchInputField = document.querySelector(".search-box input[type='text']");
+searchIcon.addEventListener("click", () => {
+  if (searchBox.classList.contains("hide-search-box")) {
+    searchBox.classList.remove("hide-search-box");
     searchInputField.value = "";
+  } else {
+    searchBox.classList.add("hide-search-box");
+  }
 });
+
+// Remove the menu and search box
+// if screen width > 550
+window.onresize = () => {
+  if (window.innerWidth > 550) {
+    if (menuContent.classList.contains("show")) {
+      menuContent.classList.add("hide");
+      menuContent.classList.remove("show");
+    }
+    if (burgerMenu.classList.contains("open")) {
+      burgerMenu.classList.remove("open");
+    }
+    searchBox.classList.add("hide-search-box");
+  }
+};
+
+// Filter By Category
+// Add event listener to the utilites box (big and medium screens)
+document
+  .querySelector(".utilities-box__content .categories")
+  .addEventListener("click", (event) => {
+    const categoryTitle = event.target.title;
+    if (categoryTitle) {
+      productsCurrentlyShown = filterByCategory(categoryTitle);
+      populateProduct(productsCurrentlyShown);
+    }
+    sortMenus.forEach((sortMenu) => {
+      sortMenu.selectedIndex = 0;
+    });
+  });
+
+// Add event listener to the categories section of the pop up menu
+// small screens
+document
+  .querySelector(".menu-content .categories")
+  .addEventListener("click", (event) => {
+    let categoryTitle = event.target.id;
+    if (!categoryTitle) {
+      categoryTitle = event.target.parentElement.id;
+    }
```

Comment:
You can use `event.currentTarget.id` instead of `event.target.id` and then you won't have to have the if-statement to look for the parentElement id - `currentTarget` will always be the element that the event listener was added to, even if a child was the actual element clicked.

Reply:
I didn't know that, Thank you